src/dataStore/DripsDataProxy.sol

```diff
@@ -0,0 +1,168 @@
+/// The data must be explicitly stored in `DripsDataStore` before its usage.
+///
+/// In some cases using the proxy is easier and cheaper than calling the driver directly.
+/// For example multisigs and governance contracts often need the whole calldata
+/// provided as a binary blob, which then sometimes needs to be stored on-chain.
+/// It's easier to build calldata consisting of a flat list of scalar parameters,
+/// and calldata is much smaller when large lists are substituted with their hashes.
+contract DripsDataProxy is Managed {
+    /// @notice The Drips contract used by this proxy.
+    Drips public immutable drips;
+    /// @notice The DripsDataStore contract used by this proxy.
+    DripsDataStore public immutable dripsDataStore;
+
+    /// @param drips_ The Drips contract to use.
+    /// @param dripsDataStore_ The DripsDataStore contract to use.
+    constructor(Drips drips_, DripsDataStore dripsDataStore_) {
+        drips = drips_;
+        dripsDataStore = dripsDataStore_;
+    }
+
+    /// @notice Receive streams from the currently running cycle from a single sender.
+    /// It doesn't receive streams from the finished cycles, to do that use `receiveStreams`.
+    /// Squeezed funds won't be received in the next calls to `squeezeStreams` or `receiveStreams`.
+    /// Only funds streamed before `block.timestamp` can be squeezed.
+    /// @param accountId The ID of the account receiving streams to squeeze funds for.
+    /// @param erc20 The used ERC-20 token.
+    /// It must preserve amounts, so if some amount of tokens is transferred to
+    /// an address, then later the same amount must be transferable from that address.
+    /// Tokens which rebase the holders' balances, collect taxes on transfers,
+    /// or impose any restrictions on holding or transferring tokens are not supported.
+    /// If you use such tokens in the protocol, they can get stuck or lost.
+    /// @param senderId The ID of the streaming account to squeeze funds from.
+    /// @param historyStartHash The sender's history hash that was valid right before
+    /// they set up the sequence of configurations described by `streamsHistoryHash`.
+    /// @param streamsHistoryHash The hash of the sequence of the sender's streams configurations,
+    /// the actual list must be stored in DripsDataStore.
+    /// It can start at an arbitrary past configuration, but must describe all the configurations
+    /// which have been used since then including the current one, in the chronological order.
+    /// Only streams described by `streamsHistoryHash` will be squeezed.
+    /// If `streamsHistoryHash` entries have no receivers, they won't be squeezed.
+    /// @return amt The squeezed amount.
+    function squeezeStreams(
+        uint256 accountId,
+        IERC20 erc20,
+        uint256 senderId,
+        bytes32 historyStartHash,
+        bytes32 streamsHistoryHash
+    ) public whenNotPaused returns (uint128 amt) {
```

Comment:
Since the underlying functions have a `whenNotPaused` modifier why here again?

Reply:
Because `Drips` will be paused only if the core protocol is broken. We need a finer-grained pausing too in case the proxy is broken. It's the same with the drivers, each of them can be paused separately in case they are broken, but the protocol itself is fine.